### DISCUSSION TITLE
Order parent and children?

### DISCUSSION THREAD

**Discussion by richcorbs** [supabase/supabase#229] [Questions] [answered]
In an `order` expression is there a way to order the parent objects and the nested/foreign/child objects? Like this:

`supabase.from("parents").select("*, children(*)").order([["parents.name", true], ["children.name", true]])`

**Comment by steve-chavez** [comment, marked as answer]
Hey @richcorbs.

Can you try doing it as:
```js
supabase
.from("parents")
.select("*, children(*)")
.order('name', { ascending: false})
.order('name', { foreignTable: 'children', ascending: false})
```
(I'm getting the example from a [postgrest-js test](https://github.com/supabase/postgrest-js/blob/3a902b3a9d3b0a765480db85b891d0b363679a76/test/resource-embedding.test.ts#L19-L25))

**Comment by kiwicopple** [comment]
If you're on the current version of `supabase-js`, it may be something like this:

```js
supabase
  .from("parents")
  .select("*, children(*)")
  .order('name.desc')
  .order('children.name.desc')
```

**Reply by richcorbs** [reply]
Looks like maybe this version isn't available via `unpkg`?

**Reply by richcorbs** [reply]
`children.name.desc` isn't working in 1.0.3:

`column parent.children.name.desc does not exist`

Maybe a regression?

**Reply by soedirgo** [reply]
Can you try doing it like Steve's example above? We changed the `order` API during the 1.0 transition.

**Reply by richcorbs** [reply]
Steve's example worked! Thank you for reminding me to try it!

**Comment by richcorbs** [comment]
Something weird is going on:

This order: `order('finishers.time_in_seconds.asc')`...

Generates this query string: `finishers.order: time_in_seconds.desc.nullslast`...

But this order: `order('finishers.time_in_seconds', {ascending: true}`...

Generates this query string: `finishers.order: time_in_seconds.asc.nullslast`...

But I get back the following finishers with time_in_seconds:

```
412
416
418
420
518
409
```
The only difference is the `409` finisher has a `registration_id` value and the others do not which I don't understand why it makes a difference in the sorting.

**Reply by kiwicopple** [reply]
We are upgrading the libaries now and I suspect you're caught between our library upgrades. I get the feeling that the old `supabase-js` has a bug in the `order()` function that we just didn't know about. Could you do us a big a favour and see if you can use your top query using this `supabase-js` library:

```
npm install github:kiwicopple/supabase-js#feature/migrate_to_typescript
```

But be aware that there are some other breaking changes in the new library: https://www.notion.so/supabase/Supabase-improved-DX-46907b88d337471791bc9b8104ecba39.

In case you can't be bothered with that - just wait a couple of weeks and we will push the new library live to NPM which has all the fixes, (tracked here: https://github.com/supabase/supabase-js/pull/50)

**Comment by richcorbs** [comment]
No matter how I try to sort finishers it sorts by inserted_at (or updated_at since they are the same).
